### PR TITLE
Show suggestion on unknown target

### PR DESCRIPTION
**Pull request by jawira**
This adds the message `Did you mean 'target'?` when target is not found.

For example:

```xml
<?xml version="1.0" encoding="UTF-8" ?>

<project name="demo" default="hello">
    <target name="setup">
        <echo>This is setup</echo>
    </target>
    <target name="hello">
        <echo>This is hello</echo>
    </target>
    <target name="clean">
        <echo>This is clean</echo>
    </target>
</project>
```

```console
$ bin/phing hell
Buildfile: /home/jawira/PhpstormProjects/fork-phing/build.xml

BUILD FAILED
 Target 'hell' does not exist in this project. Did you mean 'hello'?

Total time: 0.1012 seconds
```

This was some kind of pet feature for me, so don't hesitate to close this PR if you think it's not useful :)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -109,13 +109,15 @@ jobs:
...
           git config --global user.name "Phing Github Action"
           echo "=== RUN TESTS ==="
-          ../bin/phing -debug -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"
+          XDEBUG_MODE=coverage ../bin/phing -debug -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"
 
       - name: Transfer coverage
         if: github.repository_owner == 'phingofficial'
+        with:
+          php-version: '7.4'
         run: |
```
Comment (siad007) [inline .github/workflows/build.yml:118]: @jawira you cannot have both `with` and `run` in one step.
I would suggest to go by:
```yaml

  coverage:
    runs-on: ubuntu-latest
    needs:
      - static_code_analysis
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: '7.4'
          ini-values: xdebug.mode=coverage

      - name: Get composer cache directory
        id: composer-cache
        run: echo "::set-output name=dir::$(composer config cache-files-dir)"

      - name: Cache dependencies
        uses: actions/cache@v2
        with:
          path: ${{ steps.composer-cache.outputs.dir }}
          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
          restore-keys: ${{ runner.os }}-composer-

      - name: Test with coverage
        working-directory: test
        run: |
          echo "=== SETTING GIT IDENTITY ==="
          git config --global user.email "github-ci-build@phing.info"
          git config --global user.name "Phing Github Action"
          composer install --no-interaction --no-progress --ansi
          echo "=== RUN TESTS ==="
          ../bin/phing -debug -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"

      - name: Transfer coverage
        if: github.repository_owner == 'phingofficial'
        run: |
          bash <(curl -s https://codecov.io/bash) -f ./test/reports/clover-coverage.xml
          wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover ./test/reports/clover-coverage.xml

```

**Review by siad007 — CHANGES_REQUESTED** [review]

**Comment by siad007** [comment]
>     * `ocular.phar` is not compatible with PHP 8 (guzzle incompatiblity)

AFAIK we are using PHP 7.4 for code-coverage step?! Where exactly did you have a problem with PHP 8?

**Comment by jawira** [comment]
> > ```
> > * `ocular.phar` is not compatible with PHP 8 (guzzle incompatiblity)
> > ```
> 
> AFAIK we are using PHP 7.4 for code-coverage step?! Where exactly did you have a problem with PHP 8?

No, it wasn't PHP 7.4, by default this step used PHP 8 I added `php -v` to be sure.

**Comment by siad007** [comment]
Ah ok - I see. There was a change in the OS ubuntu-latest was updated. So yes you are right. I will change to code-coverage with PHP 8.0 in a later PR.

**Comment by siad007** [comment]
@jawira please check the new steps in my last suggestion - we have to add php 7.4 in a new single step and have the additional composer steps also (just take over my snipped)

**Comment by siad007** [comment]
There was a wrong step in my post:
```yaml
      - name: Composer install
        run: composer install --no-interaction --no-progress --ansi

      - name: Test with coverage
        working-directory: test
        run: |
          echo "=== SETTING GIT IDENTITY ==="
          git config --global user.email "github-ci-build@phing.info"
          git config --global user.name "Phing Github Action"
          echo "=== RUN TESTS ==="
          ../bin/phing -debug -Dtests.codecoverage=true -listener "phing.listener.StatisticsListener"
```

**Review by siad007 — APPROVED** [review]

**Comment by siad007** [comment]
@jawira thanks for your PR I will merge it and have a look in the build pipeline if something strange should occure in master.

**Pull request merged**